src/js/shared/nestedDataHelpers.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+export function setOrUpdateMapInMap<OuterKey, InnerKey, Value>(
+  outerMap: Map<OuterKey, Map<InnerKey, Value>>,
+  outerKey: OuterKey,
+  innerKey: InnerKey,
+  value: Value,
+): Map<OuterKey, Map<InnerKey, Value>> {
+  const innerMap = outerMap.get(outerKey) ?? new Map();
+  innerMap.set(innerKey, value);
+  if (!outerMap.has(outerKey)) {
```

Comment:
I guess it's more correct but should we just always right to the outer one, makes this logic a bit simpler to follow. idk...

Reply:
I'd lean toward leaving it as-is since it's very slightly more correct, it's hidden away in a helper anyway, and it matches the existing behavior. If you feel strongly, I can update. Thoughts?